### ISSUE TITLE
Replace current about dialog with settings page

### ISSUE THREAD

**Issue by grochocki** [microsoft/calculator#596]
**Problem Statement**
Calculator currently does not have a settings page, which will block future work. The current about dialog is also inconsistent with many other Microsoft apps.

**Evidence or User Insights**
Many suggested features have suggested adding options in settings (#538, #486, #485, #393, #382, #215), and while we have not accepted all of these pitches, it is inevitable that there will be functionality requiring settings UX.

**Proposal**
Replace current about dialog with settings page. 

**Goals**
* Replace current about dialog with settings page
* Be consistent with settings UX patterns established in other Microsoft apps
* Add basic theme selection option common across many other apps

**Non-Goals**
* Add any additional options at this time

**Low-Fidelity Concept**
![image](https://user-images.githubusercontent.com/1414423/61502861-69602380-a98a-11e9-976e-c11b66a7220a.png)

**Requested Assignment**
I'm just suggesting this idea.  I don't want to implement it.

**Comment by MicrosoftIssueBot** [comment]
This is your friendly Microsoft Issue Bot. I've seen this issue come in and have gone to tell a human about it.

**Comment by skyanchor** [comment]
[https://github.com/microsoft/calculator/issues/114](url)

**Comment by grochocki** [comment]
It looks like the pitch for #114 was accepted. I think the idea with that one is that we would just always preserve input between modes (without a setting).

**Comment by mdtauk** [comment]
![image](https://user-images.githubusercontent.com/7389110/76562664-c4a4d000-649d-11ea-8a13-8661c74fdacc.png)

![image](https://user-images.githubusercontent.com/7389110/76562679-cc647480-649d-11ea-94d1-f2aa190981a5.png)

![image](https://user-images.githubusercontent.com/7389110/76562694-d25a5580-649d-11ea-945b-eca26171eab4.png)

**Comment by grochocki** [comment]
Thanks for putting these together! They are pretty spot on with what I had in mind. A few bits of feedback:

I can't quite tell from the mockup if "Settings" is the same heading level as "App Mode", but I imagine we would want to establish clear hierarchy here (especially when we start adding more settings). For reference, many other apps leverage ```TitleTextBlockStyle``` for section headers.

Also, for consistency with other apps, the bottom section should have an "About this app" section header. We'll also want to tighten up the vertical spacing and horizontal alignment a bit in this section:

![image](https://user-images.githubusercontent.com/1414423/78107167-ce896700-73a9-11ea-9217-a9c2e9e262bc.png)
<sub>Courtesy [XAML Studio](https://www.microsoft.com/store/productId/9NTLS214TKMQ)</sub>

<details>
<summary>Generic Code Segment</summary>
```xaml
<TextBlock
    Margin="0,0,0,8"
    Style="{StaticResource TitleTextBlockStyle}"
    Text="About this app"
    TextTrimming="Clip" />

<TextBlock
    Margin="0,0,0,4"
    Style="{StaticResource BodyTextBlockStyle}"
    IsTextSelectionEnabled="True">
    <Run Text="APPNAME" />
    <Run Text="VERSION" />
    <LineBreak />
    <Run FlowDirection="LeftToRight" Text="© 2020 Microsoft. All rights reserved." />
</TextBlock>

<HyperlinkButton
    Margin="0,-2"
    HorizontalAlignment="Left"
    NavigateUri="https://go.microsoft.com/fwlink/?LinkId=529064">
    <TextBlock Text="Microsoft Software License Terms" TextWrapping="Wrap" />
</HyperlinkButton>

<HyperlinkButton
    Margin="0,-2"
    HorizontalAlignment="Left"
    NavigateUri="https://go.microsoft.com/fwlink/?LinkId=822631">
    <TextBlock Text="Microsoft Services Agreement" TextWrapping="Wrap" />
</HyperlinkButton>

<HyperlinkButton
    Margin="0,-2"
    HorizontalAlignment="Left"
    NavigateUri="https://go.microsoft.com/fwlink/?LinkID=521839">
    <TextBlock Text="Microsoft Privacy Statement" TextWrapping="Wrap" />
</HyperlinkButton>

<Button
    x:Name="FeedbackButton"
    Margin="0,24"
    Click="OnSendFeedbackButtonClick"
    Content="Send feedback" />
```
</details>

**Comment by Chips1234** [comment]
Just telling everyone, I'm working on this (and #486 since those two are similar)

**Comment by Chips1234** [comment]
And how do I change the theme?

**Comment by Chips1234** [comment]
@mdtauk How did you create that concept? Did you do it with the code? If so, would you please share the code in MainPage with me? I'm creating the spec. Thanks!

**Comment by mdtauk** [comment]
I'm afraid it's just a graphic mock up.

I did put the app into a debugger, and used the live visual tree to hide the controls to get a blank screen.

**Comment by Chips1234** [comment]
Alright. Thanks anyways.

**Comment by grochocki** [comment]
@mdtauk, I made a couple tweaks based on your mock, so I would appreciate any feedback you have on the spec.

@Chips1234, once microsoft/calculator-specs#24 is approved, it is all yours to implement, if you are still interested!

**Comment by Chips1234** [comment]
Ah, thank you!

**Comment by Chips1234** [comment]
Also, since you’re there, can you comment on my spec? Thanks

**Comment by Chips1234** [comment]
Alright. I’ll start work on this. Thanks!

**Comment by Chips1234** [comment]
The only thing is, I still can't figure out the part about the entry point: ```return mode == ViewMode::Settings;``` that code is in MainPage.cpp (since the footer is constructed there and not in NavCategory) but it does not seem to return Settings

**Comment by Chips1234** [comment]
I uploaded a branch for the full code that I'm having trouble with. Take a look here [at this branch](https://github.com/Chips1234/calculator/tree/settings-code-snippet).

**Comment by Chips1234** [comment]
@grochocki? (Sorry for the mention but I need a quick answer)

**Comment by Chips1234** [comment]
@sanderl Could you help? Thanks!

**Comment by sanderl** [comment]
@Chips1234, I've pulled the branch you have created and I'm trying out some changes. I'll let you know what I figure out.

**Comment by Chips1234** [comment]
@sanderl. Great, thanks!

**Comment by sanderl** [comment]
@Chips1234, Ok, I got it work so that it can navigate to the page. You need to update OnSettingButtonClick to assign the Model->mode value like this:

void MainPage::OnSettingsButtonClick(Object ^ sender, ItemClickEventArgs ^ e, ViewMode **/*mode*/**)
{
    **Model->Mode = ViewMode::Settings;**
}

static const list<NavCategoryInitializer> s_categoryManifest to include a new entry for settings like this (the icon needs to be updated):
 NavCategoryInitializer{ ViewMode::Settings,
                                      SETTINGS_ID,
                                      L"Settings",
                                      L"CategoryName_Settings",
                                      L"\uF515",
                                      CategoryGroupType::None,
                                      MyVirtualKey::None,
                                      nullptr,
                                      false,
                                      true } 


When I tested this out, it is navigating to a blank page which is expected because there is no content in the Settings.xaml.

**Comment by Chips1234** [comment]
Thanks for the help!

________________________________

**Comment by Chips1234** [comment]
@sanderl One last question, in SettingsPage.cpp I tried:
```cpp
void SettingsPage::ThemeRadioButtonChecked(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
{
    if (SettingsLightTheme->IsChecked)
    {
        Application.RequestedTheme = 1; 
    }
}
```
as the doc says, but it does not seem to work.

**Comment by hanzhang54** [comment]
We have replaced the about dialog with settings page in https://github.com/microsoft/calculator/pull/1618. The new Calculator supports to change the app theme without restarting the app. 😎
